libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -538,16 +485,15 @@ public new BotFrameworkAdapter Use(IMiddleware middleware)
             
 
             // Create a conversation update activity to represent the result.
-            var conversationUpdate = Activity.CreateConversationUpdateActivity();
-            conversationUpdate.ChannelId = channelId;
-            conversationUpdate.TopicName = conversationParameters.TopicName;
-            conversationUpdate.ServiceUrl = serviceUrl;
-            conversationUpdate.MembersAdded = conversationParameters.Members;
-            conversationUpdate.Id = result.ActivityId ?? Guid.NewGuid().ToString("n");
-            conversationUpdate.Conversation = new ConversationAccount(id: result.Id);
-            conversationUpdate.Recipient = conversationParameters.Bot;
-
-            using (TurnContext context = new TurnContext(this, (Activity)conversationUpdate))
+            var eventActivity = Activity.CreateEventActivity();
+            eventActivity.Name = "CreateConversation";
+            eventActivity.ChannelId = channelId;
+            eventActivity.ServiceUrl = serviceUrl;
+            eventActivity.Id = result.ActivityId ?? Guid.NewGuid().ToString("n");
+            eventActivity.Conversation = new ConversationAccount(id: result.Id);
+            eventActivity.Recipient = conversationParameters.Bot;
+
+            using (TurnContext context = new TurnContext(this, (Activity)eventActivity))
             {
                 ClaimsIdentity claimsIdentity = new ClaimsIdentity();
                 claimsIdentity.AddClaim(new Claim(AuthenticationConstants.AudienceClaim, credentials.MicrosoftAppId));
```

Comment:
I bet this needs the AppId Claim as well, that Ramjot added.

                  new Claim(AuthenticationConstants.AppIdClaim, botAppId)

Reply:
it does ?!!?

---
In reply to: [198679503](https://github.com/Microsoft/botbuilder-dotnet/pull/705#discussion_r198679503) [](ancestors = 198679503)